user-service/controller/user-controller.js

```diff
@@ -1,143 +1,66 @@
-        if (username && password) {
-            // Authenticate user
-            const signedInUser = await authenticateUser(username, password);
-            if (!signedInUser) {
-                return res.status(401).json({ message: 'User does not exist and/or wrong password.' });
-            }
-            // Create JWT
-            const token = _generateJwt(signedInUser);
-            // Send cookie
-            res.cookie('token', token, { httpOnly: true, sameSite: 'none', secure: true });
-            return res.status(200).json({ token });
-        } else {
-            return res.status(400).json({ message: 'Missing username and/or password.' });
-        }
-    } catch (err) {
-        console.log(err);
-        return res.status(500).json({ message: 'Server error when signing in user.' });
+import { 
+  checkExistence,
+  ormCreateUser as _createUser, 
+  ormUpdateUser as _updateUser, 
+  ormDeleteUser as _deleteUser,
+} from '../model/user-orm.js';
+
+export const createUser = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    if (username && email && password) {
+      // Check if user exists
+      const user = await checkExistence(username, email);
+      if (user) {
+        return res.status(401).json({ message: 'User already exists.' });
+      }
+
+      // Create user
+      await _createUser(username, email, password);
+      return res.status(201).json({ message: `Account ${username} created successfully.` });
+    } else {
+      return res.status(400).json({ message: 'Username/Email/Password is/are missing.' });
     }
-}
-
-export async function logout(req, res) {
-    const { token } = req;
-    await _blacklistJwt(token);
-    res.clearCookie('token');
-    return res.status(200).json({ message: 'Logout successful.' })
-}
-
-export async function updateAccount(req, res) {
-    const { token } = req;
-    const { newUsername, newPassword } = req.body;
-    const { username } = _decodeJwt(token);
+  } catch (_) {
+    return res.status(500).json({ message: 'Server error when creating user.' });
+  }
+};
+
+export const updateUser = async (req, res) => {
+  try {
+    const username = req.params.id; 
+    const { newPassword } = req.body;
     const newProfile = {};
-
     // Only add to the new profile to update to if the value exists
-    if (newUsername) {
-        newProfile.username = newUsername;
-    }
-
     if (newPassword) {
-        newProfile.password = newPassword;
+      newProfile.password = newPassword;
     }
-
+  
     // Directly return if newProfile is empty, no update
     if (
       newProfile && // null and undefined check
       Object.keys(newProfile).length === 0 &&
       Object.getPrototypeOf(newProfile) === Object.prototype
     ) {
-      return res.status(200).json({
-        message: 'No update on your account since no input provided.',
-      });
-    }
-
-    const updatedUser = await _updateAccount(username, newProfile);

```

Comment:
Similarly, I don't get why we discard this feature, as far as I know, we return a cookie that constitues the updated user so new cookie will be issued? We can replace the old one with this new one isn't it?

Reply:
I see, I shall help to update the UI to allow only changing password then. Thanks for clarifying!